lib/components/app/app-menu.js

```diff
@@ -35,66 +45,179 @@ class AppMenu extends Component {
       const params = qs.parse(search, { ignoreQueryPrefix: true })
       const { sessionId } = params
       if (sessionId) {
-        startOverUrl += `?${qs.stringify({sessionId})}`
+        startOverUrl += `?${qs.stringify({ sessionId })}`
       }
     }
     window.location.href = startOverUrl
   }
 
+  _togglePane = () => {
+    const { isPaneOpen } = this.state
+    this.setState({ isPaneOpen: !isPaneOpen })
+  }
+
+  _toggleSubmenu = (id) => {
+    const { expandedSubmenus } = this.state
+    const currentlyOpen = expandedSubmenus[id] || false
+    this.setState({ expandedSubmenus: { [id]: !currentlyOpen } })
+  }
+
+  _addExtraMenuItems = (menuItems) => {
+    return (
+      menuItems &&
+      menuItems.map((menuItem) => {
+        const {
+          children,
+          href,
+          iconType,
+          iconUrl,
+          id,
+          label,
+          subMenuDivider
+        } = menuItem
+        const { expandedSubmenus } = this.state
+        const isSubmenuExpanded = expandedSubmenus[id]
+
+        if (children) {
+          return (
+            <>
```

Comment:
Here, the `key` prop should be applied to the Fragment element, not the `MenuItem` (see corresp msg in browser console).

Reply:
Thanks for doing this in bb24583